### PR TITLE
reduce duplication in files

### PR DESCRIPTION
**Pull request by NathanSun1981**
In our university course, we are required to analyze an open source and figure out the technical debt. We use SonarQube to assess the code and find much duplicated code in event_mutilplexer.py and plugin_event_multiplexer.py, also event_accumulator.py and plugin_event_accumulator.py. The methods in both classes are almost same. We set the classes in event_mutilplexer and event_accumulator as superclasses, rewrote the classes in plugin_event_mulitiplexer and plugin_evetn_accumlator as subclasses for inheritance. In addition, we modified all test files for the new structure. 
Moreover, json_util.py and events_writer_manager.py contains the conditional structure that have two branches in the same "if" structure with the same implementation, so we remove the duplicated branches.

### CODE REVIEW

**Comment by NathanSun1981** [comment]
I signed it!

**Comment by chihuahua** [comment]
Thank you for taking the initiative to identify duplicate code in TensorBoard's repository and make a pull request. The reason this repository maintains parallel files containing event multiplexer logic is to maintain backwards compatibility for code within Google, so for now, I think we will decline this pull request.

This setup is suboptimal, and we will redesign TensorBoard's backend to obviate the duplicate logic in the future.

**Comment by NathanSun1981** [comment]
Thank you for your quick reply! Learning TensorBoard's architecture and code is an awesome experience for us.

File: tensorboard/backend/event_processing/plugin_event_accumulator.py [hunk 1]
```diff
@@ -148,119 +152,21 @@ def __init__(self,
...
-    self._generator = _GeneratorFromPath(path)
+    self._generator = event_accumulator._GeneratorFromPath(path)
 
     self.purge_orphaned_data = purge_orphaned_data
 
     self.most_recent_step = -1
     self.most_recent_wall_time = -1
     self.file_version = None
 
-  def Reload(self):
```
Comment (jart) [inline tensorboard/backend/event_processing/plugin_event_accumulator.py]: I like to see code deleted!

File: tensorboard/backend/event_processing/plugin_event_accumulator.py [hunk 2]
```diff
@@ -29,6 +29,9 @@
 from tensorboard.backend.event_processing import io_wrapper
 from tensorboard.backend.event_processing import plugin_asset_util
 from tensorboard.backend.event_processing import reservoir
+from tensorboard.backend.event_processing.event_accumulator import EventAccumulator
```
Comment (jart) [inline tensorboard/backend/event_processing/plugin_event_accumulator.py]: Please remove this import, per https://google.github.io/styleguide/pyguide.html#Imports

File: tensorboard/backend/event_processing/plugin_event_accumulator.py [hunk 3]
```diff
@@ -57,7 +60,8 @@
 _TENSOR_RESERVOIR_KEY = "."  # arbitrary
 
 
-class EventAccumulator(object):
+
+class EventAccumulatorPlugin(EventAccumulator):
```
Comment (jart) [inline tensorboard/backend/event_processing/plugin_event_accumulator.py]: Please do not rename this class. That would require refactoring folks who depend on it.

File: tensorboard/backend/event_processing/plugin_event_accumulator_test.py [hunk 4]
```diff
@@ -116,30 +117,30 @@ class MockingEventAccumulatorTest(EventAccumulatorTest):
   def setUp(self):
     super(MockingEventAccumulatorTest, self).setUp()
     self.stubs = tf.test.StubOutForTesting()
-    self._real_constructor = ea.EventAccumulator
-    self._real_generator = ea._GeneratorFromPath
+    self._real_constructor = ea.EventAccumulatorPlugin
```
Comment (jart) [inline tensorboard/backend/event_processing/plugin_event_accumulator_test.py]: If the class isn't renamed, the churn in this file should hopefully not be necessary. Churn makes it more difficult to `git blame`.

File: tensorboard/backend/event_processing/plugin_event_multiplexer_test.py [hunk 5]
```diff
@@ -23,9 +23,10 @@
 
 import tensorflow as tf
 
-from tensorboard.backend.event_processing import plugin_event_accumulator as event_accumulator  # pylint: disable=line-too-long
-from tensorboard.backend.event_processing import plugin_event_multiplexer as event_multiplexer  # pylint: disable=line-too-long
-
+from tensorboard.backend.event_processing import plugin_event_accumulator as ea  # pylint: disable=line-too-long
```
Comment (jart) [inline tensorboard/backend/event_processing/plugin_event_multiplexer_test.py]: I'd rather not change the history of this file to make it more succinct. I'm OK with it remaining verbose.

File: tensorboard/backend/json_util.py [hunk 6]
```diff
@@ -51,9 +51,8 @@ def Cleanse(obj, encoding='utf-8'):
   Returns:
     Unicode JSON data structure.
   """
-  if isinstance(obj, int):
```
Comment (jart) [inline tensorboard/backend/json_util.py]: Please revert this change. I special-cased `int` because it's the most likely branch. See also https://github.com/tensorflow/tensorboard/commit/c152887f924397db3c7c37f0e8d8346012e66ddc as well as

File: tensorboard/plugins/debugger/events_writer_manager.py [hunk 7]
```diff
@@ -131,13 +131,10 @@ def write_event(self, event):
 
         file_path = os.path.join(self._events_directory,
                                  self.get_current_file_name())
-        if not tf.gfile.Exists(file_path):
+        if (not tf.gfile.Exists(file_path)) or (tf.gfile.Stat(file_path).length > self._single_file_size_cap_bytes):
```
Comment (jart) [inline tensorboard/plugins/debugger/events_writer_manager.py]: No lines longer than 80 chars please.

**Comment by NathanSun1981** [comment]
we have modified all suggested code, but the CI build job still failed without error code.
